agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -550,8 +552,8 @@ func (dg *dockerGoClient) createContainer(ctx context.Context,
 	return dg.containerMetadata(ctx, dockerContainer.ID)
 }
 
-func (dg *dockerGoClient) StartContainer(ctx context.Context, id string, timeout time.Duration) DockerContainerMetadata {
-	ctx, cancel := context.WithTimeout(ctx, timeout)
+func (dg *dockerGoClient) StartContainer(ctx context.Context, id string, ctxTimeout time.Duration) DockerContainerMetadata {
```

Comment:
can you add similar comment for ctxTimeout here as well? is there valid range of values for this like it does for stop timeout?

Reply:
no, this is just what we chose for the start container timeout. we have some sane defaults here https://github.com/aws/amazon-ecs-agent/pull/1321, but since we're allowing it to be configurable from the task def we don't have a range check at this layer. the service is enforcing a lower bound of 2s and no upper bound. 

the docker daemon's start container api doesnt enforce timeouts.